### DISCUSSION TITLE
Access to the TypeParser

### DISCUSSION THREAD

**Discussion by magnetik** [CuyZ/Valinor#531] [Ideas]
Hi,

I'm using Valinor & [NelmioApiDocBundle](https://github.com/nelmio/NelmioApiDocBundle) in all my projects.

It somehow works outside the box with the default property describer from ApiDocBundle.

But when using value objects, the behavior differs  : 
* Valinor allows to give only the value when a class as a single one.
* ApiDocBundle describe this as an object with a single value.

Is there any way to "inspect" the type definition created by Valinor in order to make a NelmioApiDoc describer that fits exactly valinor type definitions? (maybe by accessing the container?)

Thanks.

**Comment by romm** [comment]
Hi @magnetik sorry for the late answer.

I do not plan to give access to the type system, as it would greatly expand the API scope and complicate the maintenance burden.

In your case (if it's not too late), I would try to expand the bundle's API, and in case of a VO retrieve the single value type and allow it in the API.

I do not see a better viable solution, hope it's ok for you. 😊
